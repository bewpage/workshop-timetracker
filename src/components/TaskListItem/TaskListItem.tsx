import React, { JSX } from 'react';
import { OperationTaskType, TaskType } from '../App/App';

const TaskListItem = ({
  operation,
}: {
  operation: OperationTaskType;
}): JSX.Element => {
  const listItem = (data: OperationTaskType): JSX.Element => (
    <li className="list-group-item d-flex justify-content-between align-items-center">
      <div id="task-operation-item-description">
        {operation.description}
        <span className="badge badge-success badge-pill ml-2"></span>
      </div>
      <div>
        <button className="btn btn-outline-success btn-sm mr-2">+15m</button>
        <button className="btn btn-outline-success btn-sm mr-2">+1h</button>
        <button className="btn btn-outline-danger btn-sm">Delete</button>
      </div>
    </li>
  );

  return <>{listItem(operation)}</>;
};

export default TaskListItem;
