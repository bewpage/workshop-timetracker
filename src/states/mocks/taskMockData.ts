import { DataState } from '../TasksContext';
import { OperationTaskType, TaskType } from '../../components/App/App';

export const taskMockData: TaskType = {
  id: '12',
  title: 'Task Title Test',
  description: 'Task Description Test',
  status: 'open',
  addedDate: '2021-01-01',
  operations: [],
};

export const operationMockData: OperationTaskType = {
  id: '1',
  description: 'Description Operation Task Test',
  timeSpent: 20,
  addedDate: '2021-01-01',
  task: taskMockData,
};

export const mockData: DataState = {
  isLoading: false,
  tasks: [taskMockData],
};

export const mockAPIData = {
  apiKey: { id: 'API_KEY_AUTH' },
  id: '12',
  title: 'Task Title Test',
  description: 'Task Description Test',
  status: 'open',
  addedDate: '2021-01-01',
};

export const mockAPIDataTest = {
  id: '12',
  title: 'Task Title Test',
  description: 'Task Description Test',
  status: 'open',
  addedDate: '2021-01-01',
};
