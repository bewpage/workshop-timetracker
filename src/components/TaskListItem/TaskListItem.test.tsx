import { render, screen } from '@testing-library/react';
import TaskListItem from './TaskListItem';
import { operationMockData } from '../../states/mocks/taskMockData';

describe('TaskListItem', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders TaskListItem component', () => {
    const { container } = render(
      <TaskListItem operation={operationMockData} />
    );

    const descriptionText = screen.getByText(
      /Description Operation Task Test/i
    );
    const deleteButton = screen.getByRole('button', { name: 'Delete' });
    const add15mButton = screen.getByRole('button', { name: '+15m' });

    expect(deleteButton).toBeInTheDocument();
    expect(add15mButton).toBeInTheDocument();
    expect(descriptionText).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
