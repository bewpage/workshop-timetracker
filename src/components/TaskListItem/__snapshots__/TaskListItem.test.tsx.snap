// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListItem renders TaskListItem component 1`] = `
<div>
  <li
    class="list-group-item d-flex justify-content-between align-items-center"
  >
    <div
      id="task-operation-item-description"
    >
      Description Operation Task Test
      <span
        class="badge badge-success badge-pill ml-2"
      />
    </div>
    <div>
      <button
        class="btn btn-outline-success btn-sm mr-2"
      >
        +15m
      </button>
      <button
        class="btn btn-outline-success btn-sm mr-2"
      >
        +1h
      </button>
      <button
        class="btn btn-outline-danger btn-sm"
      >
        Delete
      </button>
    </div>
  </li>
</div>
`;
